pkg/addon/source.go

```diff
@@ -128,9 +128,13 @@ func NewAsyncReader(baseURL, bucket, subPath, token string, rdType ReaderType) (
 		}
 		u.Path = path.Join(u.Path, subPath)
 		tp, content, err := utils.Parse(u.String())
-		if err != nil || tp != utils.TypeGithub {
+		if err != nil {
 			return nil, err
 		}
+
+		if tp != utils.TypeGithub {
+			return nil, fmt.Errorf("%s registry haven't been supported yet", tp)
+		}
```

Comment:
can you write a test about the panic case?

Reply:
fixed